cpmonitor/views.py

```diff
@@ -506,6 +508,52 @@ def markdown_uploader_view(request):
     return HttpResponse(data, content_type="application/json")
 
 
+def mstr_view(request, municipality_key):
+    url = "https://www.marktstammdatenregister.de/MaStR/Einheit/EinheitJson/GetVerkleinerteOeffentlicheEinheitStromerzeugung"
+    params = {
+        "sort": "EinheitMeldeDatum-desc",
+        "page": 1,
+        "pageSize": 10000,
+        "filter": f"Energieträger~eq~'2495'~and~Gemeindeschlüssel~eq~'{municipality_key}'",
+    }
+    try:
+        r = requests.get(url, params)
+    except:
+        return HttpResponseServerError(500)
+
+    data = r.json()["Data"]
+
+    installed_by_year = dict()
+
+    START_YEAR = 2019
+    for entry in data:
+        if not entry["InbetriebnahmeDatum"]:
+            continue
+
+        m = re.match("/Date\((-?\d*)\)/", entry["InbetriebnahmeDatum"])
+        install_date = datetime.fromtimestamp(int(m.group(1)) / 1000)
+        year = START_YEAR if install_date.year < START_YEAR else install_date.year
+
+        if not year in installed_by_year:
+            installed_by_year[year] = 0
```

Comment:
Hm I am not sure I understand this, why do we not use "Bruttoleistung" when we encounter a year for the first time?

Reply:
I am not sure I understand the question. You mean  `installed_by_year[year] = entry["Bruttoleistung"]`? You could do that, but then we would need an `else` afterwards. Otherwise we would add it twice when we first encounter a a year (see next line). Setting to 0 is just initialisation.